app/lib/Plugins/SearchEngine/Elastic8.php

```diff
@@ -31,63 +31,67 @@
  */
 
 use Elastic\Elasticsearch\Client;
+use Elastic\Elasticsearch\Exception\AuthenticationException;
 use Elastic\ElasticSearch\Exception\ClientResponseException;
 use Elastic\Elasticsearch\Exception\ElasticsearchException;
+use Elastic\Elasticsearch\Exception\MissingParameterException;
+use Elastic\Elasticsearch\Exception\ServerResponseException;
 use Monolog\Handler\ErrorLogHandler;
 use Elastic8\Mapping;
 
-require_once( __CA_LIB_DIR__ . '/Configuration.php' );
-require_once( __CA_LIB_DIR__ . '/Datamodel.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/WLPlug.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/IWLPlugSearchEngine.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/SearchEngine/BaseSearchPlugin.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8Result.php' );
+require_once(__CA_LIB_DIR__ . '/Configuration.php');
+require_once(__CA_LIB_DIR__ . '/Datamodel.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/WLPlug.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/IWLPlugSearchEngine.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/BaseSearchPlugin.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8Result.php');
 
-require_once( __CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Field.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Mapping.php' );
-require_once( __CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Query.php' );
+require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Field.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Mapping.php');
+require_once(__CA_LIB_DIR__ . '/Plugins/SearchEngine/Elastic8/Query.php');
 
 class WLPlugSearchEngineElastic8 extends BaseSearchPlugin implements IWLPlugSearchEngine {
 	# -------------------------------------------------------
-	protected $index_content_buffer = [];
+	protected array $index_content_buffer = [];
 
-	protected $indexing_subject_tablenum = null;
-	protected $indexing_subject_row_id = null;
-	protected $indexing_subject_tablename = null;
+	protected ?string $indexing_subject_tablenum = null;
+	protected ?int $indexing_subject_row_id = null;
+	protected ?string $indexing_subject_tablename = null;
 
-	/**
-	 * @var Client
-	 */
-	static protected $client;
+	static protected ?Client $client = null;
 
-	static private $doc_content_buffer = [];
-	static private $update_content_buffer = [];
-	static private $delete_buffer = [];
-	static private $record_cache = [];
+	static private array $doc_content_buffer = [];
+	static private array $update_content_buffer = [];
+	static private array $delete_buffer = [];
+	static private array $record_cache = [];
 
 	protected $elasticsearch_index_name = '';
-	protected $elasticsearch_base_url = '';
+	protected string $elasticsearch_base_url = '';
 
-	protected $version = 8;
+	protected int $version = 8;
 
 	# -------------------------------------------------------
```

Comment:
Should we make these dividers consistent? remove from here or add in other files?

Reply:
Remove them all!